apps/web/src/components/search/search-content.tsx

```diff
@@ -141,7 +141,7 @@ function Topbar() {
 function RecentSearches({ onClick }: { onClick: OnClick }) {
   const { getItems, onAdd, onRemove } = useRecentSearches();
 
-  const results = getItems();
+  const results = getItems() as Challenge[];
```

Comment:
[Explanation] `useRecentSearchesStorage` wasn't using its `<Item>` type parameter. It was effectively violating [`@typescript-eslint/no-unnecessary-type-parameters`](https://typescript-eslint.io/rules/no-unnecessary-type-parameters): the "golden rule" of generics.